Tests/SourceControlTests/GitRepositoryTests.swift

```diff
@@ -113,10 +113,21 @@ class GitRepositoryTests: XCTestCase {
     /// `Inputs`, which has known commit hashes. See the `construct.sh` script
     /// contained within it for more information.
     func testRawRepository() throws {
+#if os(Windows)
+        try XCTSkipIf(true, "test repository has non-portable file names")
+#endif
         try testWithTemporaryDirectory { path in
             // Unarchive the static test repository.
             let inputArchivePath = AbsolutePath(#file).parentDirectory.appending(components: "Inputs", "TestRepo.tgz")
-            try systemQuietly(["tar", "-x", "-v", "-C", path.pathString, "-f", inputArchivePath.pathString])
+            func executable(_ name: String) -> String {
+#if os(Windows)
+                if !name.hasSuffix(".exe") {
+                    return "\(name).exe"
+                }
+#endif
+                return name
+            }
+            try systemQuietly([executable("tar"), "-x", "-v", "-C", path.pathString, "-f", inputArchivePath.pathString])
```

Comment:
I would prefer we remove the executable func and instead do

```
#if os(Windows)
try systemQuietly(["tar.exe", "-x", "-v", "-C", path.pathString, "-f", inputArchivePath.pathString])
#else
try systemQuietly(["tar", "-x", "-v", "-C", path.pathString, "-f", inputArchivePath.pathString])
#endif
```

Reply:
🤷‍♂️